### PR TITLE
Add a scan date row in Checkmarx One Results section (AST-35838)

### PR DESCRIPTION
**Pull request by checkmarx-kobi-hagmi**
### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: src/utils/utils.ts [hunk 1]
```diff
@@ -105,7 +109,19 @@ export function getScanLabel(createdAt: string, id: string) {
     minutes +
     ":" +
```
Comment (OrShamirCM) [inline src/utils/utils.ts:110]: ??

File: src/utils/utils.ts [hunk 2]
```diff
@@ -105,7 +109,19 @@ export function getScanLabel(createdAt: string, id: string) {
...
     seconds;
-  return dateString + " " + id;
+  return dateString;
+}
+
+export function getFormattedId(label: CxScan, scanList: CxScan[]) {
+  return label === scanList[0]
```
Comment (OrShamirCM) [inline src/utils/utils.ts:116]: add validation for scan list, if there is linq use first()

File: src/views/resultsView/astResultsProvider.ts [hunk 3]
```diff
@@ -88,6 +88,9 @@ export class AstResultsProvider extends ResultsProvider {
     }
     // if there are results loaded, the tree needs to be recreated
     if (this.loadedResults !== undefined) {
+      const newItem = new TreeItem(`${getFromState(this.context, constants.scanIdKey).datetime}`, constants.calendarItem);
```
Comment (OrShamirCM) [inline src/views/resultsView/astResultsProvider.ts:91]: please have more informative vars names

File: src/utils/common/globalState.ts [hunk 4]
```diff
@@ -37,4 +37,6 @@ export class Item {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   id: string | any | undefined;
   name: string | undefined;
+  datetime: string | undefined;
+  formattedId: string | undefined;
```
Comment (OrShamirCM) [inline src/utils/common/globalState.ts:41]: I'm not sure what this means. please update this to a better name

**Review by OrShamirCM — APPROVED** [review]

File: src/utils/common/globalState.ts (hunk 4 above)
Comment (checkmarx-kobi-hagmi) [reply src/utils/common/globalState.ts]: Fixed

File: src/utils/utils.ts (hunk 2 above)
Comment (checkmarx-kobi-hagmi) [reply src/utils/utils.ts:120]: Fixed

**Pull request merged**